### ISSUE TITLE
Could not find base class "_BASE_" when creating script_templates

### ISSUE THREAD

**Issue by ICEFIR** [godotengine/godot#79236]
### Godot version

4.1.stable.mono

### System information

EndeavourOS (Arch Linux) - Godot v4.1.stable.mono  - Vulkan (Mobile) - dedicated NVIDIA GeForce RTX 3070 Ti (nvidia; 535.54.03) - AMD Ryzen 7 5800X 8-Core Processor (16 Threads)

### Issue description

**Current Situation:**
Following the [doc](https://docs.godotengine.org/en/stable/tutorials/scripting/creating_script_templates.html)
When trying to create/use script templates, gd language server returns 
```
  Failed parse script res://editor/script_templates/<Node>/default.gd
  Could not find base class "_BASE_"
```
![image](https://github.com/godotengine/godot/assets/6231162/fb44faae-45c8-4594-9b8a-a50f2c526651)

Expected:
Godot to parse _BASE_ correctly and register the script template

### Steps to reproduce

- create script_template folder
- create Node folder
- create default.gd
- write a single line: 
```
extends _BASE_
```

### Minimal reproduction project

[Reproduction Project](https://drive.google.com/file/d/1ep098NPgqmFVElGY5PdlWyCoKrIUnMMn/view?usp=drive_link)

**Comment by AThousandShips** [comment]
Does this happen if you add a line below that single line?

**Comment by ICEFIR** [comment]
> Does this happen if you add a line below that single line?

Yes, it still happens, in fact i've discovered this issue when i was attempting to use [beehave addon](https://github.com/bitbrain/beehave)
![image](https://github.com/godotengine/godot/assets/6231162/2c91be5c-5fda-4d5b-b2a5-195552731f08)

**Comment by AThousandShips** [comment]
Have you changed the setting to use `res://editor/script_templates`? Because it uses `res://script_templates` by default, as the documentation says

**Comment by ICEFIR** [comment]
ah in fact i've tested both locations in suspicion that the parser isn't reading the correct location, but neither is working.
 
In the original project i've relocated the script_templates folder to `res://editor/script_templates` to test out if things are working or not; and in the minimum reproducible project i've kept it under `res://script_templates`, and tested both importing the addon with concrete implementation & a single line. All scenario failed

**Comment by AThousandShips** [comment]
I cannot reproduce your error, opening the reproduction project doesn't cause any errors, nor does creating a script from the template

**Comment by ICEFIR** [comment]
:thinking: opening the project indeed does not cause any errors, but its only when i am opening the script i am getting this error from the language server 

1 sec, let me try another platform

**Comment by AThousandShips** [comment]
I don't think there's any direct checks for the templates like that

See the note:
> The script templates have the same extension as the regular script files. This may lead to an issue of a script parser treating those templates as actual scripts within a project. To avoid this, make sure to ignore the directory containing them by creating an empty .gdignore file. The directory won't be visible throughout the project's filesystem anymore, yet the templates can be modified by an external text editor anytime.

**Comment by AThousandShips** [comment]
At least it's expected that it will show an error when opening it in the editor, as `_BASE_` is not a class

**Comment by ICEFIR** [comment]
> The script templates have the same extension as the regular script files. This may lead to an issue of a script parser treating those templates as actual scripts within a project. To avoid this, make sure to ignore the directory containing them by creating an empty .gdignore file. The directory won't be visible throughout the project's filesystem anymore, yet the templates can be modified by an external text editor anytime.

Ah i see - i think this is where the issue lies; in the [original addon project](https://github.com/bitbrain/beehave/tree/godot-4.x/script_templates) there is indeed `.gdignore` file - however the issue is that i am still getting the error from the language server (which might be an issue with VSCode plugin itself?)

**Comment by AThousandShips** [comment]
When do you get this error?

**Comment by ICEFIR** [comment]
Ah when i open the godot project in VSCode (As i finds copilot X super useful :) ), using godot-tools and C# Tools for Godot

**Comment by AThousandShips** [comment]
You are still opening it via the language server, so the note above still applies, so it's probably on the side of VSCode, or possibly the language server providing those scripts, so editing these scripts are not guaranteed or intended to work, and the warnings are expected, but might be some issue with that it connects via the script server incorrectly, but I'm not sure if the issue is on the Godot side or if the editor simply doesn't respect the `.gdignore`

**Comment by ICEFIR** [comment]
Okay so to sum it up, the conclusion here is that, at the moment, if editors (be it external or not) sends godot editor extension script to godot language server, this error will be an expected response. Hence its up to 3rd party plugin to correctly ignore these files (by honoring .gdignore)

whether or not this behavior should persist forward is up to debate.  

> There could be an argument for adding special treatment for this, but that would add a lot of complexity to the editor that I don't think is worth it

I guess i could take a look - though i dont have too much experience in either language implementation or godot code base yet.
Maybe in the future :) though you are right on that, this may or may not be worth it given it provides very little benefit for the moment. 

> We could emphasise further that these templates aren't supposed to be edited in the built-in editor (if that is indeed the case)

What i feel is that emphasizing on documentation does not establish a very strong contract, given the document is fairly lengthy and its easy to glance over details as this. it would be better to establish an automatics mechanism to deal with these issues instead of manual processes

> We could also automatically exclude the configured templates directory without a specific .gdignore
:thing i do feel that ideally gd language server can automatically honor .gdignore configuration..... not too sure how hard it would be though

**Comment by raulsntos** [comment]
I may be missing something but the user seems to be using GDScript. But if the issue is about C#, then it would be a duplicate of:
- https://github.com/godotengine/godot/issues/60703.

**Comment by ICEFIR** [comment]
Ah its a different issue - this is specifically for GDScript. The C# part is a side note

**Comment by limbonaut** [comment]
Same with _TS_ tokens.
Perhaps, Godot shouldn't parse files in the "res://script_template/**" location. Allow editing templates in the script editor, but don't parse them at all.

**Comment by FabriceCastel** [comment]
Is there any legitimate use case where you'd want Godot to parse files in a subdirectory of `res://script_templates/`? I can't imagine there would be, so ignoring them as @limbonaut suggest is how I would've expected this to behave in the first place 🤔

**Comment by SnrsGu** [comment]
I also ran into this problem, but when I ignored it and created the script normally with this template, it worked fine
